src/main/java/com/example/filevault/service/DataBaseStorageService.java

```diff
@@ -0,0 +1,209 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Service
+@RequiredArgsConstructor
+public class DataBaseStorageService implements FileService {
+
+    private final DateBaseFileRepository fileRepository;
+    Path rootLocation = Paths.get(FileVaultConstants.STORAGE_LOCATION);
+
+    {
+        try {
+            Files.createDirectories(rootLocation);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public FileEntity createEntity(String name, String type, String comment, String contentFolderPath, int size) {
+        return FileEntity.builder()
+                .id(UUID.randomUUID())
+                .name(name)
+                .uploadDate(new Date())
+                .modifiedDate(new Date())
+                .comment(comment)
+                .contentFolderPath(contentFolderPath)
+                .size(size)
+                .extension(type)
+                .build();
+    }
+
+    @Override
+    public FileDto upload(MultipartFile file, String comment) {
+        if (FileSizeUtils.toMB(file.getSize()) >= FileVaultConstants.MAX_FILE_SIZE_MB)
+            throw new TooLargeFileSizeException("File Size Cant be more than " + FileVaultConstants.MAX_FILE_SIZE_MB + "MB");
+
+        String fullFileName = file.getOriginalFilename();
+        String fileName = FilenameUtils.getNameWithoutExtension(fullFileName);
+        String fileExtension = FilenameUtils.getExtension(fullFileName);
+        int fileSize;
+        try {
+            fileSize = file.getBytes().length;
+        } catch (IOException e) {
+            throw new CantReadFileContentException(".getBytes() method fails", e);
+        }
+
+        FileEntity newEntity = createEntity(fileName, fileExtension, comment, rootLocation.toString(), fileSize);
+        Path destinationFile = rootLocation.resolve(newEntity.getId().toString() + '.' + fileExtension);
+
+        try (InputStream inputStream = file.getInputStream()) {
+            Files.copy(inputStream, destinationFile, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new RuntimeException(e); // TODO: create custom exception
+        }
+
+        return FileDto.of(fileRepository.save(newEntity));
+    }
+
+    @Override
+    public List<FileDto> getAll() {
+        return fileRepository.findAll()
+                .stream()
+                .map(FileDto::of)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public FileBytesAndNameById getBytesAndNameById(UUID id) {
+        Optional<FileEntity> optionalFile = fileRepository.findById(id);
+        if (optionalFile.isEmpty())
+            throw new FileNotFoundException("File not found :(");
```

Comment:
fileRepository.findById(id).orElseThrow(() ->new FileNotFoundException("File not found :("));

Reply:
ok